latch/ldata/path.py

```diff
@@ -0,0 +1,171 @@
+from pathlib import Path
+from typing import Generator, Optional, Union
+from urllib.parse import urljoin
+from uuid import uuid4
+
+import gql
+from latch_sdk_config.latch import NUCLEUS_URL
+from latch_sdk_gql.execute import execute
+
+from latch.ldata.node import (
+    LDataNodeType,
+    LDataPerms,
+    PermLevel,
+    get_node_data,
+    get_node_metadata,
+    get_node_perms,
+)
+from latch.ldata.transfer import download, remote_copy, upload
+from latch.ldata.transfer.progress import Progress
+from latch_cli.tinyrequests import post
+from latch_cli.utils import get_auth_header, urljoins
+from latch_cli.utils.path import is_remote_path
+
+
+class LPath:
+    def __init__(self, path: str):
+        if not is_remote_path(path):
+            raise ValueError(f"Invalid LPath: {path} is not a Latch path")
+        self.path = path
+        self._node_id = None
+
+    @property
+    def node_id(self) -> str:
+        if self._node_id is None:
+            self._node_id = get_node_data(self.path).data[self.path].id
+        return self._node_id
+
+    @property
+    def exists(self) -> bool:
```

Comment:
using `exists` i almost always the wrong thing. imo this API should not exist at all

you almost always want to try doing something instead and catch exceptions

this is because `exists` + something opens you to a time-of-check time-of-use issue